### PR TITLE
WIP: PSA SE driver mocks

### PR DESCRIPTION
**Pull request by Patater**
A rebase of https://github.com/ARMmbed/mbed-crypto/pull/164/ on the latest API branch.

Not for review yet.

### CODE REVIEW

File: tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function [hunk 1]
```diff
@@ -0,0 +1,553 @@
+/* BEGIN_HEADER */
+#include "psa_crypto_helpers.h"
+#include "psa/crypto_se_driver.h"
+
+#include "psa_crypto_se.h"
+
+/** The minimum valid lifetime value for a secure element driver. */
+#define MIN_DRIVER_LIFETIME 2
```
Comment (AndrzejKurek) [inline tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function]: This and the next define are unused.

Comment (Patater) [reply tests/suites/test_suite_psa_crypto_se_driver_hal_mocks.function]: Same as #164. Will fix.

**Comment by Patater** [comment]
Rebased to remove not needed defines

**Comment by Patater** [comment]
Using this branch as a base for hook work. This PR not needed anymore.

**Pull request closed without merging**